src/eredis.erl

```diff
@@ -226,13 +226,11 @@ create_multibulk(Args) ->
 to_bulk(B) when is_binary(B) ->
     [<<$$>>, integer_to_list(iolist_size(B)), <<?NL>>, B, <<?NL>>].
 
-%% @doc: Convert given value to binary. Fallbacks to
-%% term_to_binary/1. For floats, throws {cannot_store_floats, Float}
-%% as we do not want floats to be stored in Redis. Your future self
-%% will thank you for this.
+%% @doc: Convert given value to binary. Fallbacks to term_to_binary/1.
 to_binary(X) when is_list(X)    -> list_to_binary(X);
 to_binary(X) when is_atom(X)    -> atom_to_binary(X, utf8);
 to_binary(X) when is_binary(X)  -> X;
 to_binary(X) when is_integer(X) -> integer_to_binary(X);
-to_binary(X) when is_float(X)   -> throw({cannot_store_floats, X});
+to_binary(X) when is_float(X)   -> list_to_binary(io_lib_format:fwrite_g(X));
+%% OTP-25: to_binary(X) when is_float(X)   -> float_to_binary(X, [short]);
```

Comment:
That's correct, Elixir releases are pinned to support a specific OTP version.

Reply:
Oh and yes, you're right that scientific notation is fine, that was my mistake and I noted it in a later comment :)

Reply:
Updated to use `io_lib:format` with the "~.20g" format string ✔️ 

Adding an OTP_RELEASE condition will also need the same branching during tests, so it can match the new "short" format in OTP 25. I'll think I can add it if you want, but it's your burden to maintain over time :) Using only `io_lib:format` is good enough for me, and it's predictably 20 decimal digits.

Reply:
In the tests, can't we just compare the result as a float? There's no need to check the textual representation, is there?

Reply:
I see tests are already comparing RESP protocol representation, so I guess we should keep it like that. Well, I guess we can live with an OTP_RELEASE condition also in the tests.

Another idea: How about using `io_lib_format:fwrite_g(X)` for OTP < 25 and `short` for OTP >= 25? It will not break in future versions. :-)